File: src/main/java/com/github/nsc/de/shake/parser/node/factor/IntegerNode.java
package com.github.nsc.de.shake.parser.node.factor;

import com.github.nsc.de.shake.parser.node.ValuedNode;
import com.github.nsc.de.shake.util.characterinput.position.PositionMap;

public class IntegerNode extends ValuedNode {

    private final int number;

    public IntegerNode(PositionMap map, int number) {
        super(map);
        this.number = number;
    }

    public int getNumber() {
        return number;
    }

    @Override
    public String toString() {
        return String.valueOf(number);
    }
}

File: parser/src/main/kotlin/com/github/nsc/de/shake/parser/node/factor/IntegerNode.kt
package com.github.nsc.de.shake.parser.node.factor

import com.github.nsc.de.shake.parser.node.ValuedNode
import com.github.nsc.de.shake.util.characterinput.position.PositionMap

class IntegerNode(map: PositionMap, val number: Int) : ValuedNode(map) {
    override fun toString(): String = number.toString()
}